cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -262,6 +267,10 @@ type LocalEtcd struct {
 	// command line except without leading dash(es).
 	ExtraArgs map[string]string
 
+	// ExtraEnvs is an extra set of environment variables to pass to the control plane component.
+	// +optional
+	ExtraEnvs []v1.EnvVar
```

Comment:
Comments are not updated. https://github.com/kubernetes/kubernetes/pull/118867/files#diff-a919b4d8ff1a3b03bb5691a0725e0683ff5e3fa4e12c4f81690ed46b156816d7R290 too,

Overall LGTM.

Reply:
@pacoxu this is etcd, there is no such proxy env passed to etcd compared with `ControlPlaneComponent`, and we haven't override anything like what we does for `ControlPlaneComponent`.

Reply:
https://github.com/kubernetes/kubernetes/pull/118867/files#diff-509876ddb9971f8787c885617ebaabe5da81755c5f1cddc1e2877af0d47928b2R227

Reply:
/hold

please add it everywhere. doesn't matter if we skip proxy.

actually why are we not adding proxy envs to etcd? isn't that a bug?

Reply:
Generally etcd will not access outside, only peers should be accessed by etcd. And only local apiserver can access etcd.

For apiserver, it may access webhooks. Scheduler also can have webhook.

Reply:
hm, maybe a peer would not be able to reach a peer because of proxy.

Reply:
there is no such a request for passing the proxy env so far, we can add it when there is a clear usecase,  and this is not a API change, it will be easy to address then.

Reply:
added the comment for all the occurrences! @pacoxu @neolit123 looking for LGTM again.

Reply:
A corner case that is nice to support😄. Either is OK with me for this point.